kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -138,15 +139,17 @@ private PartitionedTableId getRecordTable(SinkRecord record) {
     TableId baseTableId = topicsToBaseTableIds.get(record.topic());
 
     PartitionedTableId.Builder builder = new PartitionedTableId.Builder(baseTableId);
-    if (useMessageTimeDatePartitioning) {
-      if (record.timestampType() == TimestampType.NO_TIMESTAMP_TYPE) {
-        throw new ConnectException(
-            "Message has no timestamp type, cannot use message timestamp to partition.");
-      }
-
-      builder.setDayPartition(record.timestamp());
-    } else {
-      builder.setDayPartitionForNow();
+    if(useIngestionTimePartitioning) {
```

Comment:
This might be wrong. BigQuery support 3 types of partitioning. `useMessageTimeDatePartitioning`
is for Date/timestamp so we should be able to use `MessageTimeDatePartitioning` when `useIngestionTimePartitioning = False`.
```
You can partition BigQuery tables by:
Ingestion time: Tables are partitioned based on the data's ingestion (load) date or arrival date.
Date/timestamp: Tables are partitioned based on a TIMESTAMP or DATE column.
Integer range: Tables are partitioned based on an integer column.```

Reply:
This has nothing to do with BigQuery's partitioning types. It's more of whether the table name should have the decorator appended. I have renamed the config parameter to reflect the same.

Reply:
With or without partition decorator is type of ingestion time partition.

Reply:
Te configuration just says whether it should add a decorator or not. It has got nothing to do with partitioning strategy.